src/Processors/Transforms/WindowTransform.cpp

```diff
@@ -987,9 +985,16 @@ void WindowTransform::writeOutCurrentRow()
             // FIXME does it also allocate the result on the arena?
             // We'll have to pass it out with blocks then...
 
-            /// We should use insertMergeResultInto to insert result into ColumnAggregateFunction
-            /// correctly if result contains AggregateFunction's states
-            a->insertMergeResultInto(buf, *result_column, arena.get());
+            if (ws.is_aggregate_function_state)
```

Comment:
I guess a similar improvement might be possible in other places where `insertMergeResultInto` is called in a non constexpr environment right? I see FunctionInitializeAggregation and FunctionArrayReduce

Reply:
Adding a commit improving:
* FunctionInitializeAggregation -> `Select initializeAggregation('uniqExact', number) from numbers(100_000_000) settings max_threads=1 format Null;` around 20% faster
* ArrayReduce: Less improvement, around 10% on a query like: `Select arrayReduce('sum', a) from a settings max_threads=1 format Null;`

@kitaisreal WDYT?

Reply:
Yes, make sense. We need to merge this.